ee/clickhouse/queries/column_optimizer.py

```diff
@@ -79,8 +79,23 @@ def should_query_elements_chain_column(self) -> bool:
                 if uses_elements_chain(entity.get_action()):
                     return True
 
+        # PathFilters use inferred properties
+        if isinstance(self.filter, PathFilter):
+            if (
```

Comment:
Re above comment - this business logic could just be embedded into `filter.entities` or be otherwise elsewhere. This doesn't belong in column_optimizer.